src/lib/components/Swap/SwapPropValidator.tsx

```diff
@@ -0,0 +1,78 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { DefaultAddress, SwapProps } from 'lib/components/Swap'
+import { IntegrationError } from 'lib/errors'
+import { PropsWithChildren, useEffect, useMemo, useState } from 'react'
+
+import { isAddress } from '../../../utils'
+
+function isAddressOrAddressMap(addressOrMap: DefaultAddress): boolean {
+  if (typeof addressOrMap === 'object') {
+    return Object.values(addressOrMap).every((address) => isAddress(address))
+  }
+  if (typeof addressOrMap === 'string') {
+    return typeof isAddress(addressOrMap) === 'string'
+  }
+  return false
+}
+
+export default function SwapPropValidator(props: PropsWithChildren<SwapProps>) {
+  // convenience fee constraints
+  const [convenienceFeeChecked, setConvenienceFeeChecked] = useState(false)
+  const { convenienceFee, convenienceFeeRecipient } = props
+  useEffect(() => {
+    setConvenienceFeeChecked(false)
+    if (convenienceFee) {
+      if (convenienceFee > 100 || convenienceFee < 0) {
+        throw new IntegrationError('Set widget convenienceFee to at least 400px.')
+      }
+      if (!convenienceFeeRecipient) {
+        throw new IntegrationError('convenienceFeeRecipient is required when convenienceFee is set.')
+      }
+      const MustBeValidAddressError = new IntegrationError('convenienceFeeRecipient must be a valid address.')
```

Comment:
Instead of reusing the error, can you include more context in the two error cases' messages?